metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,592 @@
+            if not self.config.keyspace_pattern.allowed(keyspace_name):
+                self.report.report_dropped(keyspace_name)
+                continue
+
+            # 1. Construct and emit the container aspect.
+            yield from self._generate_keyspace_container(keyspace_name)
+
+            # 2. get all tables for this keyspace and emit their metadata (dataset, container and schema info)
+            try:
+                yield from self._extract_tables_from_keyspace(keyspace_name)
+            except Exception as e:
+                logger.warning(
+                    f"Failed to extract table metadata for keyspace {keyspace_name}",
+                    exc_info=True,
+                )
+                self.report.report_warning(
+                    "keyspace",
+                    f"Exception while extracting keyspace tables {keyspace_name}: {e}",
+                )
+
+            # 3. get all views for this keyspace and emit their metadata (dataset, container and lineage info)
+            try:
+                yield from self._extract_views_from_keyspace(keyspace_name)
+            except Exception as e:
+                logger.warning(
+                    f"Failed to extract view metadata for keyspace {keyspace_name}",
+                    exc_info=True,
+                )
+                self.report.report_warning(
+                    "keyspace",
+                    f"Exception while extracting keyspace views {keyspace_name}: {e}",
+                )
+
+    def _generate_keyspace_container(
+        self, keyspace_name: str
+    ) -> Iterable[MetadataWorkUnit]:
+        keyspace_container_key = self._generate_keyspace_container_key(keyspace_name)
+        yield from gen_containers(
+            container_key=keyspace_container_key,
+            name=keyspace_name,
+            sub_types=[DatasetContainerSubTypes.KEYSPACE],
+        )
+
+    def _generate_keyspace_container_key(self, keyspace_name: str) -> ContainerKey:
+        return KeyspaceKey(
+            keyspace=keyspace_name,
+            platform=self.platform,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
+
+    # get all tables for a given keyspace, iterate over them to extract column metadata
+    def _extract_tables_from_keyspace(
+        self, keyspace_name: str
+    ) -> Iterable[MetadataWorkUnit]:
+        tables = self.cassandra_session.execute(GET_TABLES_QUERY, [keyspace_name])
+        tables = sorted(
+            tables,
+            key=lambda t: getattr(
+                t, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"]
+            ),
+        )  # sorted so output is consistent
+        for table in tables:
+            # define the dataset urn for this table to be used downstream
+            table_name: str = getattr(
+                table, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"]
+            )
+            dataset_name: str = f"{keyspace_name}.{table_name}"
```

Comment:
given we're linking datasets to their keyspace containers should i just name it by the table name?